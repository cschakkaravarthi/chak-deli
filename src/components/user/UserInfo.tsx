import React, { FC } from 'react';
import get from 'lodash.get';
import UserInfoModel, { Phone } from '../../shared/models/UserInfo.model';

type Props = {
  usersInfo?: UserInfoModel;
};

const UserInfo: FC<Props> = props => {
  const { usersInfo } = props;

  if (!usersInfo || !usersInfo.networkID) return null;

  return (
    <>
      <div className='user__info'>
        <div className='user__name'>
          <a href='#'>
            {usersInfo.lastName}, {usersInfo.firstName}
          </a>
        </div>
        <div className='user__title'>
          <p>{usersInfo.jobTitle}</p>
          <p>{usersInfo.department}</p>
        </div>
        <div className='user__email'>
          <a type='email' href={`mailto:${usersInfo.email}`}>
            {usersInfo.email}
          </a>
        </div>
        {get(usersInfo, 'location.office') && (
          <div className='user__office'>
            <p>Office: {get(usersInfo, 'location.office', '')}</p>
          </div>
        )}
        <div className='user__address'>
          <p>
            {get(usersInfo, 'location.room', '')} <br />
            {get(usersInfo, 'location.address', '')} <br />
            {`${get(usersInfo, 'location.city', '')}, ${get(usersInfo, 'location.state', '')} `}
            {get(usersInfo, 'location.postalCode', '')}
          </p>
        </div>
        <div className='user__phone'>
          {get(usersInfo, 'phone', [] as Phone[]).map((item: Phone, index: number) => (
            <p key={index}>
              {item.type === 'Desk' && <span>Phone: {item.num}</span>}
              {item.type === 'Mobile' && <span>Mobile: {item.num}</span>}
            </p>
          ))}
        </div>
      </div>
    </>
  );
};

export default UserInfo;
