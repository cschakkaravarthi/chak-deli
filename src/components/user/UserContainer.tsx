import React, { FC } from 'react';
import get from 'lodash.get';
import UserInfo from './UserInfo';
import ManagerInfo from './ManagerInfo';
import ProfilePicture from './ProfilePicture';
import DirectReportInfo from './DirectReportInfo';

import UserInfoModel from '../../shared/models/UserInfo.model';
import images from '../../images/images';

type Props = {
  history?: any;
  userDetails?: UserInfoModel;
};

const UserContainer: FC<Props> = props => {
  const { history, userDetails = {} } = props;

  if (!userDetails || !userDetails.networkID) return null;

  return (
    <>
      {get(userDetails, 'error') ? (
        <p>{get(userDetails, 'message')}</p>
      ) : (
        <div className='user'>
          <div className='umgc-layout--two-column'>
            <div className='layout-main'>
              <div className='user__content'>
                <div className='user__item'>
                  <div className='user-info__container'>
                    <ProfilePicture profilePictureSrc={get(userDetails, 'photo.medium', images.defaultAvatar)} />
                    <UserInfo usersInfo={userDetails} />
                  </div>
                </div>
                <div className='user__item'>
                  <ManagerInfo history={history} reportsTo={userDetails.reportsTo} />
                  <DirectReportInfo history={history} directReports={userDetails.directReports} />
                </div>
              </div>
            </div>
          </div>
        </div>
      )}
    </>
  );
};

export default UserContainer;
