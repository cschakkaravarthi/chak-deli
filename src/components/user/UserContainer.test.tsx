import React from 'react';
import TestRenderer from 'react-test-renderer';
import UserContainer from './UserContainer';

import { dummyUserInfo } from '../../shared/models/UserInfo.model';

describe('UserContainer', () => {
  it('renders correctly without info', () => {
    const tree = TestRenderer.create(<UserContainer />).toJSON();
    expect(tree).toBe(null);
  });

  it('renders correctly with info', () => {
    const tree = TestRenderer.create(<UserContainer userDetails={dummyUserInfo} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
