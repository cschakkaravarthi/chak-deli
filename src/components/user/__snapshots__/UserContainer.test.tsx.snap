// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserContainer renders correctly with info 1`] = `<p />`;
